src/vmm/src/vm.rs

```diff
@@ -0,0 +1,105 @@
+use crate::vcpu::{self, Vcpu};
+use kvm_bindings::{kvm_pit_config, kvm_userspace_memory_region, KVM_PIT_SPEAKER_DUMMY};
+use kvm_ioctls::{Kvm, VmFd};
+use std::sync::Arc;
+use std::thread;
+use vm_device::device_manager::IoManager;
+use vmm_sys_util::eventfd::EventFd;
+
+/// A KVM specific implementation of a Virtual Machine.
+///
+/// Provides abstractions for working with a VM. Once a generic Vm trait will be available,
+/// this type will become on of the concrete implementations.
+pub struct KvmVm {
+    fd: VmFd,
```

Comment:
I'll have to think a bit about this one. I don't particularly like the approach because it is leaking implementation details about the VM in the devices, and I am thinking about maybe finding a better abstraction there.

Reply:
There are kinda two separate issues here. Holding the fd as an `Arc<VmFd>` and providing access from the outside is an overall useful interface addition IMO. On the device side, right now the top level devices also make some specific assumptions (i.e. we're using `kvm` and things like `EventFd`s), and then they also incorporate some building blocks which are already shaping up to be generic. Also, I guess this second thing is something we can talk about on the device PR.